x-pack/plugins/session_view/public/components/SessionView/index.tsx

```diff
@@ -89,22 +107,186 @@ export const SessionView = ({ sessionEntityId, height }: SessionViewDeps) => {
+              <MonacoEditor
+                height={400}
+                language="javascript"
+                options={{
+                  lineNumbers: 'on',
+                  language: 'json',
+                  readOnly: true,
+                }}
+                value={JSON.stringify(selectedProcess?.events || {}, null, 4)}
+              />
+              <EuiSpacer size="xxl" />
+            </>
+          )}
+          <EuiTitle size="s">
+            <span>Session Detail</span>
+          </EuiTitle>
+          <EuiSpacer />
+          <MonacoEditor
+            height={400}
+            language="javascript"
+            options={{
+              lineNumbers: 'on',
+              language: 'json',
+              readOnly: true,
+            }}
+            value={JSON.stringify(data?.[0]?.process.session || {}, null, 4)}
+          />
+          <EuiSpacer size="xxl" />
+          <EuiTitle size="s">
+            <span>Server Detail</span>
+          </EuiTitle>
+          {/* Add server detail */}
+          <EuiSpacer size="xxl" />
+          <EuiTitle size="s">
+            <span>Alert Detail</span>
+          </EuiTitle>
+          {/* Add alert detail conditionally */}
+        </>
+      ),
+    },
+  ];
+
   if (!data.length) {
     return renderNoData();
   }
 
   return (
     <>
-      <EuiSearchBar query={searchQuery} onChange={onSearch} />
-      <div css={styles.processTree}>
-        <ProcessTree
-          sessionEntityId={sessionEntityId}
-          forward={data}
-          searchQuery={searchQuery}
-          selectedProcess={selectedProcess}
-          onProcessSelected={onProcessSelected}
-        />
-      </div>
+      <EuiFlexGroup>
+        <EuiFlexItem>
+          <EuiSearchBar query={searchQuery} onChange={onSearch} />
+        </EuiFlexItem>
+        <EuiFlexItem grow={false}>
+          <EuiButton onClick={toggleDetailPanel} iconType="list" fill>
+            Detail Panel
+          </EuiButton>
+        </EuiFlexItem>
+      </EuiFlexGroup>
+      <EuiSplitPanel.Outer
+        direction="row"
+        color="transparent"
+        borderRadius="none"
+        css={styles.outerPanel}
+      >
+        <EuiSplitPanel.Inner paddingSize="none" css={styles.treePanel}>
+          <div css={styles.processTree}>
+            <ProcessTree
+              sessionEntityId={sessionEntityId}
+              forward={data}
+              searchQuery={searchQuery}
+              selectedProcess={selectedProcess}
+              onProcessSelected={onProcessSelected}
+            />
+          </div>
+        </EuiSplitPanel.Inner>
+        {isDetailOpen && (
+          <EuiSplitPanel.Inner
+            paddingSize="s"
+            color="plain"
+            css={isDetailMounted ? styles.detailPanelIn : styles.detailPanelOut}
+            onAnimationEnd={() => {
+              if (!isDetailMounted) setIsDetailOpen(false);
+            }}
+          >
+            <EuiTabs>
+              {detailPanelTabs.map((tab, idx) => (
```

Comment:
So I kind of had a different idea about the tabs or "sections" for details.. Was thinking more like the process.events array would be the tabs e.g

![image](https://user-images.githubusercontent.com/16198204/142294509-71b0f342-ca55-4e58-81da-3e244c15e376.png)
Using an accordion in this example.. but yea, each section would just render that nice json explorer (with the collapsing ability). What do you guys think? Since each process is many events, this would be the most non obfuscated way of informing the user that there are multiple events at play....

Reply:
It's almost like a sub timeline for each process....

Reply:
alternatively we merge those 3 events objects together in order and show 1 json
e.g  const combined = { ...forkEvent, ...execEvents, ...exitEvent};

but that might be confusing and misleading.

Reply:
Yea I think merging the 3 could be misleading. I do think having an accordion to just show the json explorer is better. I can work on that. Do we need a section for session details tho?